crates/services/txpool_v2/src/pool.rs

```diff
@@ -236,9 +255,23 @@ where
         &mut self,
         constraints: Constraints,
     ) -> Vec<ArcPoolTx> {
-        self.selection_algorithm
-            .gather_best_txs(constraints, &mut self.storage)
+        let start = std::time::Instant::now();
+        let metrics = self.config.metrics;
+        let best_txs = self
+            .selection_algorithm
+            .gather_best_txs(constraints, &mut self.storage);
+
+        if metrics {
+            Self::record_select_transaction_time_in_nanoseconds(start)
+        };
+
+        best_txs
             .into_iter()
+            .inspect(|storage_data| {
+                if metrics {
```

Comment:
If you move this "If" before `best_txs.into_iter()` it will be more performant and compiler can do better optimizations

Reply:
First approach: https://github.com/FuelLabs/fuel-core/pull/2385/commits/fec31bb864b8b55069d3504e380b2a47d43084f1

But I actually ended up with [this](https://github.com/FuelLabs/fuel-core/pull/2385/commits/27d03216a91d36b35fb0a2f347015042957a19fe), trading code clarity for the potential double iteration if metrics are enabled.